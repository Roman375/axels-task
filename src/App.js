import React, { useEffect } from 'react'
import { Route, Switch } from 'react-router'
import { connect } from 'react-redux'

import {Container} from './styled/styled'
import { UsersList, OverviewPage } from './components'
import { getEmployee } from './redux/ducks/employees'
import { getOverviews } from './redux/ducks/overviews'

const App = ({getEmployee, getOverviews, ...props}) => {

  const getOverview = async (employeeName) => {
    await getOverviews(employeeName)
  }

  useEffect(() => {
   getEmployee()
   
  }, [getEmployee])

  return (
    <Container>
      <Switch>
      <Route exact path='/' render={() => 
      <UsersList employees={props.employees} getOverview={getOverview}/>}/>
      <Route exact path='/overview/:employee?' render={() => 
      <OverviewPage overview={props.overviews} />}/>
      </Switch>
    </Container>
  )
}

const mapStateToPrors = (state) => {
  return{
    employees: state.employees,
    overviews: state.overviews
  }
}

export default connect(mapStateToPrors, {getEmployee, getOverviews})(App)
