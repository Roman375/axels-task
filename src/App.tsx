import React, { FC } from 'react'
import { connect } from 'react-redux'

import { getEmployee } from './redux/ducks/employees'
import { getOverviews } from './redux/ducks/overviews'
import { HomePage } from './components'

type Props = {
  getEmployee: () => void
  getOverviews: (employeeName: string) => void
  employees: {
    employees: string[]
  }
  overviews: Array<object>
}

type mapStateToProrsType = {
  employees: {
    employees: []
  }
  overviews: Array<object>
}

const App: FC<Props> = (props) => <HomePage {...props}/>

const mapStateToPrors = (state: mapStateToProrsType) => ({
  employees: state.employees,
  overviews: state.overviews,
})

export default connect(mapStateToPrors, { getEmployee, getOverviews })(App)
