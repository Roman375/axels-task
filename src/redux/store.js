import { applyMiddleware, compose, createStore } from 'redux';
import createSagaMiddleware from 'redux-saga'

import reducer from './reducers'
import {watchedEmployees, watchedOverviews} from './sagas'

const sagaMiddleware = createSagaMiddleware()
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(reducer, composeEnhancers(applyMiddleware(sagaMiddleware)) )

sagaMiddleware.run(watchedEmployees)
sagaMiddleware.run(watchedOverviews)

export default store