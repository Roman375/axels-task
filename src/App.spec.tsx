import renderer from 'react-test-renderer'
import App from './App'
import { Provider } from 'react-redux'

import store from './redux/store'

describe('<App />', () => {
  const props = {
    getEmployee: () => {},
    getOverviews: () => {},
    employees: [],
    overviews: [],
  }

  describe('With Snapshot Testing', () => {
    it('Render component App', () => {
      const component = renderer.create(
        <Provider store={store}>
          <App {...props} />
        </Provider>
      )
      expect(component).toMatchSnapshot()
    })
  })
})
