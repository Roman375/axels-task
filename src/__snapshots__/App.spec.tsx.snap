// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> With Snapshot Testing Render component App 1`] = `
<section
  className="sc-bdfBwQ iVrWnZ"
>
  <form
    className="sc-hKgILt hBPRLS"
  >
    <input
      aria-label="Search"
      className="form-control me-2"
      onChange={[Function]}
      placeholder="Search"
      type="search"
    />
  </form>
  <section
    className="sc-eCssSg fuiPTt row"
  >
    <h1
      className="sc-gKsewC lcXnrY"
    >
      No Employees
    </h1>
  </section>
</section>
`;
